controllers/agentmachine_controller.go

```diff
@@ -364,7 +373,7 @@ func (r *AgentMachineReconciler) findAgent(ctx context.Context, log logrus.Field
 }
 
 func getAgentMachineRefLabel(agentMachine *capiproviderv1.AgentMachine) string {
-	return string(agentMachine.GetUID())
+	return agentMachine.Name
```

Comment:
Do we need to consider this change for existing agent machines on upgrade?

Reply:
Yes good point, I think it would be a good idea to relabel them (if it doesn't exist or has the wrong value)

Reply:
Modified to always ensure this label!